apps/web/frontend/components/taskList/taskCell.tsx

```diff
@@ -0,0 +1,68 @@
+import { ErrorMessage } from '@hookform/error-message'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { useForm } from 'react-hook-form'
+import { BiTrash } from 'react-icons/bi'
+import { CgSpinner } from 'react-icons/cg'
+
+import { Button, InputField } from '@progwise/timebook-ui'
+
+import { TaskFragment, useTaskUpdateMutation } from '../../generated/graphql'
+import { TaskFormData, taskInputSchema } from './taskList'
+
+interface TaskCellProps {
+  task: TaskFragment
+  canDelete?: boolean
+  onDelete?: () => void
+}
+
+export const TaskCell = ({ task, onDelete, canDelete }: TaskCellProps) => {
+  const [{ fetching }, taskUpdate] = useTaskUpdateMutation()
+  const {
+    setError,
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<TaskFormData>({
+    mode: 'onChange',
+    defaultValues: {
+      title: task.title,
+    },
+    resolver: yupResolver(taskInputSchema),
+  })
+
+  const handleSubmitTask = async (taskData: TaskFormData) => {
+    const result = await taskUpdate({
+      id: task.id,
+      data: {
+        projectId: task.project.id,
+        title: taskData.title,
+      },
+    })
```

Comment:
I think it's good that the update got in here.

@WINt1703, what do you think about putting the deletion in here too?